### ISSUE TITLE
BUG: `DataFrame.rank` & `Series.rank` results are inconsistent

### ISSUE THREAD

**Issue by galipremsagar** [pandas-dev/pandas#43310]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
>>> import pandas as pd
>>> import numpy as np
>>> values = [-np.inf, 0, np.inf, np.nan, 2, np.nan]         
>>> s = pd.Series(values)
>>> s
0   -inf
1    0.0
2    inf
3    NaN
4    2.0
5    NaN
dtype: float64
>>> kwargs = {'method': 'dense', 'na_option': 'bottom', 'ascending': False, 'pct': False, 'numeric_only': False}
>>> s.rank(**kwargs)
0    4.0
1    3.0
2    1.0
3    5.0
4    2.0
5    5.0
dtype: float64
>>> pd.DataFrame({'a':s}).rank(**kwargs)
     a
0  4.0
1  3.0
2  1.0
3  4.0
4  2.0
5  4.0
>>> 

```

#### Problem description

The results being returned by `Series.rank` & `DataFrame.rank` seem to be inconsistent.

#### Expected Output

Output of `DataFrame.rank` must be returned correctly(as is being done by `Series.rank`)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 5f648bf1706dd75a9ca0d29f26eadfbb595fe52b
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.11.0-27-generic
Version          : #29~20.04.1-Ubuntu SMP Wed Aug 11 15:58:17 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.2
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 57.4.0
Cython           : 0.29.24
pytest           : 6.2.4
hypothesis       : 6.17.2
sphinx           : 4.1.2
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.27.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 2021.07.0
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.53.1

</details>

**Comment by phofl** [comment]
Duplicate of #32593

**Comment by galipremsagar** [comment]
@phofl This issues seems to be occurring in the most recent version of pandas as posted in the versions tab.

**Comment by phofl** [comment]
This works on master

**Comment by phofl** [comment]
But you are correct, this was fixed by #41931

**Comment by galipremsagar** [comment]
Thanks @phofl !
